### PR TITLE
Add CI workflow to lint and check formatting of Go code

### PR DESCRIPTION
**Pull request by MatteoPologruto**
On every push and pull request that affects relevant files, check the repository's [Go](https://golang.org/) module for:

- Common detectable errors in the code.
- Use of outdated APIs
- Code style violations
- Code formatting inconsistency
- Misconfiguration

### CODE REVIEW

**Comment by MatteoPologruto** [comment]
I'll fix the Check Go errors in another PR, then rebase this one.

File: .github/workflows/check-go-task.yml [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+    needs: run-determination
+    if: needs.run-determination.outputs.result == 'true'
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+
+      matrix:
+        module:
+          # TODO: add paths of all Go modules here
```
Comment (umbynos) [inline .github/workflows/check-go-task.yml:174]:
```suggestion
```
leftover of the copy paste

File: .github/workflows/check-go-task.yml [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
```
Comment (umbynos) [inline .github/workflows/check-go-task.yml:211]:
```suggestion
```
same here

**Review by umbynos — CHANGES_REQUESTED** [review]

File: .github/workflows/check-go-task.yml [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
```
Comment (umbynos) [inline .github/workflows/check-go-task.yml:66]:
```suggestion
```

File: .github/workflows/check-go-task.yml [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
```
Comment (umbynos) [inline .github/workflows/check-go-task.yml:100]:
```suggestion
```

File: .github/workflows/check-go-task.yml [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
```
Comment (umbynos) [inline .github/workflows/check-go-task.yml:137]:
```suggestion
```

**Review by umbynos — APPROVED** [review]
Good work :checkered_flag:

**Pull request merged**